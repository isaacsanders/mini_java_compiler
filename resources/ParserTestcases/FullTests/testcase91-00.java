/* This is a test. */
class Test {
	public static void main (String[] args) {
		System.out.println(2 + 13); // cool
	}
}

class Evil {
	//Near as I can tell this is valid minijava...
	public Evil Evil(){
		Evil evil = new Evil();
		return evil;//its funny how weird this word starts to look
	}
	
	public int good(int good, /* vs */ int evil){
		return good---------------evil;
	}

}
	

	
