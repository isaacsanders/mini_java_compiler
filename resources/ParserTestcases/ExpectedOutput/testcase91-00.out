StmtList ::= 
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr PLUS Expr
Stmt ::= SYSTEMOUTPRINTLN LPAREN Expr RPAREN SEMI
StmtList ::= StmtList Stmt
MainClassDecl ::= ...
ClassDeclList ::= 
OptExtends ::= 
ClassVarDeclList ::= 
MethodDeclList ::= 
Type ::= ID
OptFormalList ::= 
StmtList ::= 
Type ::= ID
CallExpr ::= NEW ID LPAREN RPAREN
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
MethodDecl ::= ... 
MethodDeclList ::= MethodDeclList MethodDecl
Type ::= INT
Type ::= INT
Formal ::= Type ID
FormalList ::= Formal
Type ::= INT
Formal ::= Type ID
FormalList ::= FormalList COMMA Formal
OptFormalList ::= FormalList
StmtList ::= 
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
Expr ::= UnaryExpr
Expr ::= Expr MINUS Expr
MethodDecl ::= ... 
MethodDeclList ::= MethodDeclList MethodDecl
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
Program ::= MainClassDecl ClassDeclList
